tests/test_multi_async.py

```diff
@@ -178,3 +178,45 @@ def test_012_ifup_node4():
     assert node4.wait_until_state(target_state="secondary")
     assert node1.wait_until_state(target_state="primary")
     assert node2.wait_until_state(target_state="secondary")
+
+def test_013_drop_node4():
+    node4.destroy()
+
+    print()
+    assert node1.wait_until_state(target_state="primary")
+    assert node2.wait_until_state(target_state="secondary")
+    assert node3.wait_until_state(target_state="secondary")
+
+def test_014_fail_node1():
+    node1.fail()
+
+    assert node2.wait_until_state(target_state="wait_primary")
+    assert node3.wait_until_state(target_state="secondary")
+
+    ssn = ""
+    eq_(node2.get_synchronous_standby_names(), ssn)
+    eq_(node2.get_synchronous_standby_names_local(), ssn)
+
+def test_015_stop_primary():
+    node2.fail()
+
+    print()
+    assert node3.wait_until_state(target_state="report_lsn")
```

Comment:
I don't think this should happen. It should stay a `secondary` (or should become `catchingup` maybe).
This causes some weird behavior as a side effect. When node2 comes up now, without node1 coming up as well, suddenly node2 becomes `primary` instead of staying `wait_primary`.

Reply:
This also seems to mess up some other things. When node1 comes up after what I described above, then it will stay in demoted forever.

Reply:
Okay I'm going to see about adding another series of tests with node1 restarted first then. I still thing REPORT_LSN is a good assignment for the non-candidate standby when there's no other node available, because that's the state that we have for a standby without a any upstream node at the moment.